src/shared_utilities/FileReader.jl

```diff
@@ -0,0 +1,413 @@
+"""
+    FileReader
+
+This module coordinates reading of input data from NetCDF files,
+and allows us to read in data containing a time component.
+It also includes regridding and temporal interpolations
+of this data.
+
+This is based on ClimaCoupler.jl's BCReader and TimeManager modules.
+"""
+module FileReader
+
+using ClimaComms
+using ClimaCore: Fields, Spaces
+using Dates
+using JLD2
+import CFTime
+
+using ClimaLSM.Regridder
+
+export to_datetime,
+    datetime_to_strdate,
+    FileInfo,
+    FileState,
+    SimInfo,
+    PrescribedData,
+    prescribed_data_init,
+    read_data_fields!,
+    next_date_in_file,
+    interpolate_data
+
+
+"""
+    FileInfo
+
+Stores information about the current data being read in from a file.
+
+# Inputs:
+- regrid_dirpath::String     # directory for storing files used in regridding
+- outfile_root::String       # root for files generated when reading data from input file
+- varname::String            # name of the variable we're reading from the input file
+- all_dates::V               # vector containing all dates of the input file
+- date_idx0::I               # index of the first data in the file being used for this simulation
+"""
+struct FileInfo{V, I}
+    regrid_dirpath::String
+    outfile_root::String
+    varname::String
+    all_dates::V
+    date_idx0::I
+end
+
+"""
+    FileState
+
+Stores information about the current data being read in from a file.
+
+# Inputs:
+- data_fields::F        # tuple of two fields at consecutive dates, that will be used for interpolation
+- date_idx::I           # index of the data in the file currently being used
+- segment_length::I     # length of each date segment (used in the daily interpolation)
```

Comment:
maybe "# length of the time interval between the two data field entries; used in temporal interpolation"

Reply:
(not daily in general)